src/NLog/LayoutRenderers/ExceptionDataLayoutRenderer.cs

```diff
@@ -0,0 +1,135 @@
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.LayoutRenderers
+{
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Text;
+    using NLog.Common;
+    using NLog.Config;
+    using NLog.Internal;
+
+    /// <summary>
+    /// Exception information provided through 
+    /// a call to one of the Logger.*Exception() methods.
+    /// </summary>
+    [LayoutRenderer("exceptiondata")]
+    [LayoutRenderer("exception-data")]
+    [ThreadAgnostic]
+    [ThreadSafe]
+    public class ExceptionDataLayoutRenderer : LayoutRenderer, IRawValue
+    {
+        private string _dataKey;
+        /// <summary>
+        /// Gets or sets the Data Key 
+        /// </summary>
+        [DefaultParameter]
+        public string DataKey
+        {
+            get => _dataKey;
+
+            set
+            {
+                _dataKey = SanitizeDataKey(value);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExceptionDataLayoutRenderer" /> class.
+        /// </summary>
+        public ExceptionDataLayoutRenderer()
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets whether to render innermost Exception from <see cref="Exception.GetBaseException()"/>
+        /// </summary>
+        [DefaultValue(false)]
+        public bool BaseException { get; set; }
+
+        /// <inheritdoc />
+        bool IRawValue.TryGetRawValue(LogEventInfo logEvent, out object value)
+        {
+            value = GetTopException(logEvent);
+            return true;
+        }
+
+        private Exception GetTopException(LogEventInfo logEvent)
+        {
+            return BaseException ? logEvent.Exception?.GetBaseException() : logEvent.Exception;
+        }
+
+        /// <inheritdoc />
+        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
+        {
+            Exception primaryException = GetTopException(logEvent);
+            if (primaryException != null)
+            {
+                AppendDataItem(builder, primaryException, DataKey);
```

Comment:
Think you should remove `protected virtual AppendDataItem` and move the implementation here.

Consider adding a `string Format { get; set; }`-property and use it like this:

            var value = ex.Data[dataKey];
            if (value != null)
            {
                var formatProvider = GetFormatProvider(logEvent);
                builder.AppendFormattedValue(value, Format, formatProvider, ValueFormatter);
            }